core/storage/btree.rs

```diff
@@ -4228,6 +4231,113 @@ impl BTreeCursor {
             _ => false,
         }
     }
+
+    /// Count the number of entries in the b-tree
+    ///
+    /// Only supposed to be used in the context of a simple Count Select Statement
+    pub fn count(&mut self) -> Result<CursorResult<usize>> {
+        if self.count == 0 {
+            self.move_to_root();
+        }
+
+        if let Some(_mv_cursor) = &self.mv_cursor {
+            todo!("Implement count for mvcc");
+        }
+
+        let mut mem_page_rc;
+        let mut mem_page;
+        let mut contents;
+
+        loop {
+            mem_page_rc = self.stack.top();
+            return_if_locked_maybe_load!(self.pager, mem_page_rc);
+            mem_page = mem_page_rc.get();
+            contents = mem_page.contents.as_ref().unwrap();
+
+            /* If this is a leaf page or the tree is not an int-key tree, then
+             ** this page contains countable entries. Increment the entry counter
+             ** accordingly.
+             */
+            // TODO: (pedrocarlo) does not take into account if "tree is not an int-key tree" condition
```

Comment:
if we won't handle this now, then perhaps assert that page type is either tableleaf or tableinterior.

however, why not just handle the index case here already? doing the `count()` on an index is faster in theory because indexes generally hold less data than tables, meaning less page I/O.

Reply:
basically `if !matches(contents.page_type(), PageType::TableInterior` you should count the cells